spring-boot-project/spring-boot-docker-compose/src/main/java/org/springframework/boot/docker/compose/service/connection/postgres/PostgresEnvironment.java

```diff
@@ -44,11 +46,19 @@ class PostgresEnvironment {
 	}
 
 	private String extractPassword(Map<String, String> env) {
+		if (hasTrustAuthMethod(env)) {
+			return null;
+		}
 		String password = env.getOrDefault("POSTGRES_PASSWORD", env.get("POSTGRESQL_PASSWORD"));
 		Assert.state(StringUtils.hasLength(password), "PostgreSQL password must be provided");
 		return password;
 	}
 
+	private Boolean hasTrustAuthMethod(Map<String, String> env) {
+		String postgresAuthMethod = env.getOrDefault("POSTGRES_HOST_AUTH_METHOD", "password");
+		return Objects.equals(postgresAuthMethod, "trust");
```

Comment:
We don't really use `Objects.equals`. This could be `return "trust".equals(postgresAuthMethod);` instead.